### DISCUSSION TITLE
useProxies: 'never' still throws error on IE11

### DISCUSSION THREAD

**Discussion by filipengberg** [mobxjs/mobx#2607] [General] [answered]
To make MobX 6 work in IE11 I have added the following code at the beginning of my index.tsx:

```
import { configure } from 'mobx';
configure({ useProxies: 'never' });
```

**Intended outcome:**

The app should work in IE11 without any Proxy errors from Mobx.

**Actual outcome:**

I get the following error from mobx:

```
`Proxy` objects are not available in the current environment. Please configure MobX to enable a fallback implementation.`
```
**How to reproduce the issue:**

Tried creating a reproducible app in CodeSandbox but it won't run at all in IE11 :(

I did some debugging and it seems that mobx runs the following code before `configure` has been run:

https://github.com/mobxjs/mobx/blob/50b4eb50fde8f08a9e9156af238834164ba23773/packages/mobx/src/api/observable.ts#L157

This means that if you have an observable array in your app mobx will try to create it using `createObservableArray` instead of `createLegacyArray` since `globalState.useProxies` has not yet been set to false.

**Versions**

```
"mobx": "^6.0.3",
"mobx-react-lite": "^3.1.5"
```

**Comment by mweststrate** [comment, marked as answer]
This means you are not calling `configure` soon enough. Note that `import` statements of observable using modules will already execute the module body, so you might need to reorder your configure and import statements a bit more, or make sure that your top level statements don't initialize observables yet.The stack trace should reveal where you are trying to create an observable array .

**Comment by filipengberg** [comment]
You’re absolutely right, My store containing observables was initialized when imported which caused the code to run before configure. I refactored my store to be initialized later and now it works!

Thanks a lot for pointing me in the right direction and sorry for creating a false bug report! 🙂

**Comment by vertic4l** [comment]
We've called `configure` soon enough, but this stopped working in IE11 and we don't know why.

<img width="574" alt="Bildschirmfoto 2022-01-07 um 16 03 56" src="https://user-images.githubusercontent.com/1771763/148562937-c15e484d-ddad-4ca3-9ec0-79b890cd5d09.png">

**Reply by vertic4l** [reply]
Moving import of mobx + configure at the very beginning of the file seems to help.
